components/list/list-item.js

```diff
@@ -268,37 +252,55 @@ class ListItem extends RtlMixin(LitElement) {
 	}
 
 	render() {
-
-		const label = this.selectable ? html`<label class="d2l-list-item-label" for="${this._checkBoxId}" aria-labelledby="${this._contentId}"></label>` : null;
-		const link = this.href ? html`<a class="d2l-list-item-link" href="${ifDefined(this.href)}" aria-labelledby="${this._contentId}"></a>` : null;
-		const beforeContent = this.selectable
-			? html`<input id="${this._checkBoxId}" class="d2l-input-checkbox" @change="${this._handleCheckboxChange}" type="checkbox" .checked="${this.selected}" ?disabled="${this.disabled}"><slot name="illustration"></slot>`
-			: html`<slot name="illustration"></slot>`;
-
 		const classes = {
-			'd2l-list-item-container': true,
-			'd2l-list-item-flex': label || link,
-			'd2l-visible-on-ancestor-target': true
+			'd2l-visible-on-ancestor-target': true,
+			'focusing': this._focusing,
 		};
 		const contentClasses = {
 			'd2l-list-item-content': true,
-			'd2l-list-item-content-extend-separators': this._extendSeparators
+			'd2l-list-item-content-extend-separators': this._extendSeparators,
+			'hovering': this._hoveringLink,
+			'focusing': this._focusingLink,
 		};
 
 		return html`
-			<div class="${classMap(classes)}" data-breakpoint="${this._breakpoint}">
-				${label}
-				${link}
-				<div id="${this._contentId}"
+			<d2l-list-item-generic-layout
+				@focusin="${this._handleFocusIn}"
+				@focusout="${this._handleFocusOut}"
+				class="${classMap(classes)}"
+				data-breakpoint="${this._breakpoint}"
+				data-separators="${ifDefined(this._separators)}"
+				?grid-active="${this.role === 'rowgroup'}">
+				${ this.draggable ? html`
+				<div slot="outside-control"></div>
+				` : nothing }
+				${this.selectable ? html`
+				<div slot="control">
+					<input id="${this._checkBoxId}" class="d2l-input-checkbox" @change="${this._handleCheckboxChange}" type="checkbox" .checked="${this.selected}" ?disabled="${this.disabled}">
+				</div>
+				<div slot="control-action"></div>
+				` : nothing }
+				${ this.href ? html`
+				<a slot="content-action"
+					href="${this.href}"
+					aria-labelledby="${this._contentId}"
+					@mouseenter="${this._handleMouseEnterLink}"
+					@mouseleave="${this._handleMouseLeaveLink}"
+					@focus="${this._handleFocusLink}"
+					@blur="${this._handleBlurLink}"></a>
+				` : nothing }
+				<div slot="content"
 					class="${classMap(contentClasses)}"
-					data-separators="${ifDefined(this._separators)}">
-					<div class="d2l-list-item-content-flex">
-						${beforeContent}
-						<div class="d2l-list-item-main"><slot></slot></div>
-						<slot name="actions"></slot>
-					</div>
+					id="${this._contentId}">
+					<slot name="illustration"></slot>
+					<slot></slot>
 				</div>
-			</div>
+
+				<div class="d2l-list-item-actions-container" slot="actions">
+					<slot name="actions"></slot>
+				</div>
+			</d2l-list-item-generic-layout>
+			<div class="d2l-list-item-active-border"></div>
```

Comment:
I don't love this, but we need it for displaying a bottom border. Normally the bottom border is overlapped by the lower item's top border.